qiskit/optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -13,42 +13,48 @@
 # that they have been altered from the originals.
 
 """A wrapper for minimum eigen solvers from Aqua to be used within the optimization module."""
-from copy import deepcopy
 from typing import Optional, Any, Union, Tuple, List
 
 import numpy as np
-from qiskit.aqua.algorithms import MinimumEigensolver
+from qiskit.aqua.algorithms import MinimumEigensolver, MinimumEigensolverResult
 from qiskit.aqua.operators import StateFn, DictStateFn
 
 from .optimization_algorithm import OptimizationAlgorithm, OptimizationResult
 from ..converters.quadratic_program_to_qubo import QuadraticProgramToQubo
 from ..problems.quadratic_program import QuadraticProgram, Variable
 
 
-class MinimumEigenOptimizerResult(OptimizationResult):
+class MinimumEigenOptimizationResult(OptimizationResult):
     """ Minimum Eigen Optimizer Result."""
 
-    def __init__(self, x: Union[List[float], np.ndarray], fval: float, variables: List[Variable],
+    def __init__(self, x: Union[List[float], np.ndarray], fval: float,
+                 variables: List[Variable],
                  samples: List[Tuple[str, float, float]],
-                 qubo_converter: QuadraticProgramToQubo) -> None:
+                 eigensolver_result: Optional[MinimumEigensolverResult] = None) -> None:
         """
         Args:
             x: the optimal value found by ``MinimumEigensolver``.
             fval: the optimal function value.
             variables: the list of variables of the optimization problem.
             samples: the basis state as bitstring, the QUBO value, and the probability of sampling.
-            qubo_converter: ``QuadraticProgram`` to QUBO converter.
+            eigensolver_result: the result obtained from the underlying algorithm.
         """
-        super().__init__(x=x, fval=fval, variables=variables,
-                         raw_results={'samples': samples, 'qubo_converter': qubo_converter})
+        super().__init__(x, fval, variables, None)
+        self._samples = samples
+        self._eigensolver_result = eigensolver_result
 
     @property
     def samples(self) -> List[Tuple[str, float, float]]:
-        """ returns samples """
-        return self._raw_results['samples']
+        """Returns samples."""
+        return self._samples
+
+    @property
+    def eigensolver_result(self) -> MinimumEigensolverResult:
```

Comment:
In the optimizer constructor in this module the param is named `min_eigen_solver`. Would this be better (bit longer for sure) as `min_eigen_solver_result`. Aqua also has a EigenSolver which can provide more than the minimum one back so the name needs to have min(imum) in it I think, and if it matches the constructor param then it will tie these together better.

Reply:
I thought it is pretty long but agree this makes sense. So, renamed to `min_eigen_solver_result`.